### DISCUSSION TITLE
button_card_templates.yaml # 'variables' not usable for entities

### DISCUSSION THREAD

**Discussion by legraml** [matt8707/hass-config#54] [Q&A]
Hi there, first of all let me express my biggest thank you for all the effort you take in keeping this project alive! 

I follow your updates regularly and struggled with updating the button_card_template.yaml file.

I set up light groups in light.yaml and when I updated button_card_template.yaml to use light-popup-cards, I got the error message 'you have to define an entity id" which the light group obviously has. So I found out, that I would have to replace "variables.entity_id" with "entity.entity_id" to get everything up and running. I am wondering, why you changed that to "variables" and where the assignment takes place in the process?

Thanks for helping me.

**Comment by matt8707** [comment]
here

https://github.com/matt8707/hass-config/blob/3949eb9caaecedcf49e9e6eca02be653643805e2/button_card_templates.yaml#L37

(if entity is defined return entity.entity_id)
